hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -228,8 +234,27 @@ private ClosableIterator<T> makeBootstrapBaseFileIterator(HoodieBaseFile baseFil
       if (start != 0) {
         throw new IllegalArgumentException("Filegroup reader is doing bootstrap merge but we are not reading from the start of the base file");
       }
+      PartitionPathParser partitionPathParser = new PartitionPathParser();
+      Object[] partitionValues = partitionPathParser.getPartitionFieldVals(partitionPathFields, partitionPath, readerContext.getSchemaHandler().getTableSchema());
+      // filter out the partition values that are not required by the data schema
+      Object[] filteredPartitionValues = new Object[0];
+      Option<String[]> filteredPartitionPathFields = Option.empty();
+      if (partitionPathFields.isPresent()) {
+        Schema dataSchema = dataFileIterator.get().getRight();
+        List<String> fields = new ArrayList<>();
+        List<Object> values = new ArrayList<>();
+        for (int i = 0; i < partitionPathFields.get().length; i++) {
+          String field = partitionPathFields.get()[i];
+          if (dataSchema.getField(field) != null) {
+            fields.add(field);
+            values.add(partitionValues[i]);
+          }
+        }
+        filteredPartitionPathFields = fields.isEmpty() ? Option.empty() : Option.of(fields.toArray(new String[0]));
+        filteredPartitionValues = values.toArray(new Object[0]);
+      }
```

Comment:
nit: if only `filteredPartitionPathFields` and `filteredPartitionValues` are needed, the data iterator's schema can be passed to `PartitionPathParser` for parsing to return a list of field names and values that are needed for merging bootstrap readers.

Reply:
The parser requires the table schema because it needs to know how to parse the fields for a given type. For example, a date can be multiple folders in the path

Reply:
Got it.  That is OK.  Wondering if the filtering logic can be put into the parser or a method so the logic here looks concise.  If it cannot be put into the parser easily, we can leave it as is.